service/tablet_allocator.cc

```diff
@@ -730,12 +730,15 @@ class load_balancer {
             plan.merge(std::move(dc_plan));
         }
 
-        // Make plans for repair jobs
-        plan.set_repair_plan(co_await make_repair_plan(plan));
-
         // Merge table-wide resize decisions, may emit new decisions, revoke or finalize ongoing ones.
+        // Note : Resize plans should be generated before repair plans to avoid scheduling repairs when there is pending resize finalization
         plan.merge_resize_plan(co_await make_resize_plan(plan));
 
+        if (plan.resize_plan().finalize_resize.empty()) {
+            // Make plans for repair jobs iff there are no pending resize finalizations
```

Comment:
nit: The comment doesn't add any new information here, it expresses in words what the if statement already means.

Reply:
ack. I have updated the comment.